### ISSUE TITLE
Task - Default avatar is not matching on confirm task page for the same account

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22358]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open a new account (Account A) and copy the email
2. Open another new account (Account B) and click on FAB > Assign task
3. Give a title and click next
4. Click on assignee
5. Paste the copied email (Account A)
6. Compare the default avatar of the Assignee and the share somewhere

## Expected Result:
Both default avatars should be the same since it is the same account

## Actual Result:
Different default avatars are being shown

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.36-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/a4c3e176-a577-4df2-aba7-9f9fc5d6ab7b

https://github.com/Expensify/App/assets/93399543/52ec12ba-8d87-445b-9e63-6e56bfaf5a7f

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Nathan-Mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688448710392939

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by thienlnam** [comment]
Adding hold on this because I think it will be resolved by this internal issue https://github.com/Expensify/Expensify/issues/294203

**Comment by JmillsExpensify** [comment]
Still holding

**Comment by JmillsExpensify** [comment]
Hold hold

**Comment by JmillsExpensify** [comment]
Hold

**Comment by JmillsExpensify** [comment]
Hold

**Comment by JmillsExpensify** [comment]
Same

**Comment by JmillsExpensify** [comment]
Related issue is actually good to go, so removing the hold on this one.

**Comment by JmillsExpensify** [comment]
@kavimuru Are we able to have QA test this one and see if it's still reproducible?

**Comment by Nathan-Mulugeta** [comment]
Yes, I am able to reproduce this again. I will drop down the steps to follow to reproduce this issue: 

1. Open a new account 
2. Click on FAB > New chat > Enter an email 
3. Reload the page 
4. Click on FAB > Assign task > Give the task a title > Click next
5. Click on the assignee option and paste the email address you started a chat with on step 2
6. Compare the avatars of the assignee field and the share somewhere field

I will attach a screen recording below this: 


https://github.com/Expensify/App/assets/111440031/6e49e678-77e9-478e-9571-f0837c17eb5b

**Comment by thienlnam** [comment]
Does this still happen if the other account replies to your DM? I think this is expected right now because we're storing the accountID optimistically (until the other user responds)

**Comment by Nathan-Mulugeta** [comment]
If the other account replies to my DM nothing happens at first but if I manually reload the page and then do the repro steps again the issue will no longer be there.

**Comment by thienlnam** [comment]
Gotcha, yeah I think in this case this bug is expected then due to our optimistic accountIDs

**Comment by thienlnam** [comment]
Closing since this is expected
